### PR TITLE
fix: flaky test for api/v1/targets/metadata.

### LINKED ISSUE #6431: web/api/v1: Unit Tests fail nondeterministically on master
## Bug Report
**What did you do?**
* Checkout latest master branch (as of commit 0ae048ff ).
* Run `go test ./web/api/v1` multiple times, without test caching.

**What did you expect to see?**
The test should always pass.
**What did you see instead? Under which circumstances?**
Sometimes the test fails. It might need multiple runs to reproduce this.

```

--- FAIL: TestEndpoints (0.03s)
    --- FAIL: TestEndpoints/local (0.01s)
        api_test.go:1157: run 0	GET	"query=2&time=123.4"
        api_test.go:1157: run 0	POST	"query=2&time=123.4"
        api_test.go:1157: run 1	GET	"query=0.333&time=1970-01-01T00%3A02%3A03Z"
        api_test.go:1157: run 1	POST	"query=0.333&time=1970-01-01T00%3A02%3A03Z"
        api_test.go:1157: run 2	GET	"query=0.333&time=1970-01-01T01%3A02%3A03%2B01%3A00"
        api_test.go:1157: run 2	POST	"query=0.333&time=1970-01-01T01%3A02%3A03%2B01%3A00"
        api_test.go:1157: run 3	GET	"query=0.333"
        api_test.go:1157: run 3	POST	"query=0.333"
        api_test.go:1157: run 4	GET	"end=2&query=time%28%29&start=0&step=1"
        api_test.go:1157: run 4	POST	"end=2&query=time%28%29&start=0&step=1"
        api_test.go:1157: run 5	GET	"end=2&query=time%28%29&step=1"
        api_test.go:1157: run 5	POST	"end=2&query=time%28%29&step=1"
        api_test.go:1157: run 6	GET	"query=time%28%29&start=0&step=1"
        api_test.go:1157: run 6	POST	"query=time%28%29&start=0&step=1"
        api_test.go:1157: run 7	GET	"end=2&query=time%28%29&start=0"
        api_test.go:1157: run 7	POST	"end=2&query=time%28%29&start=0"
        api_test.go:1157: run 8	GET	"query=invalid%5D%5Bquery&time=1970-01-01T01%3A02%3A03%2B01%3A00"
        api_test.go:1157: run 8	POST	"query=invalid%5D%5Bquery&time=1970-01-01T01%3A02%3A03%2B01%3A00"
        api_test.go:1157: run 9	GET	"end=100&query=invalid%5D%5Bquery&start=0&step=1"
        api_test.go:1157: run 9	POST	"end=100&query=invalid%5D%5Bquery&start=0&step=1"
        api_test.go:1157: run 10	GET	"end=2&query=time%28%29&start=1&step=0"
        api_test.go:1157: run 10	POST	"end=2&query=time%28%29&start=1&step=0"
        api_test.go:1157: run 11	GET	"end=1&query=time%28%29&start=2&step=1"
        api_test.go:1157: run 11	POST	"end=1&query=time%28%29&start=2&step=1"
        api_test.go:1157: run 12	GET	"end=1489667272.372&query=time%28%29&start=148966367200.372&step=1"
        api_test.go:1157: run 12	POST	"end=1489667272.372&query=time%28%29&start=148966367200.372&step=1"
        api_test.go:1157: run 13	GET	"match%5B%5D=test_metric2"
        api_test.go:1157: run 13	POST	"match%5B%5D=test_metric2"
        api_test.go:1157: run 14	GET	"match%5B%5D=test_metric1%7Bfoo%3D~%22.%2Bo%22%7D"
        api_test.go:1157: run 14	POST	"match%5B%5D=test_metric1%7Bfoo%3D~%22.%2Bo%22%7D"
        api_test.go:1157: run 15	GET	"match%5B%5D=test_metric1%7Bfoo%3D~%22.%2Bo%24%22%7D&match%5B%5D=test_metric1%7Bfoo%3D~%22.%2Bo%22%7D"
        api_test.go:1157: run 15	POST	"match%5B%5D=test_metric1%7Bfoo%3D~%22.%2Bo%24%22%7D&match%5B%5D=test_metric1%7Bfoo%3D~%22.%2Bo%22%7D"
        api_test.go:1157: run 16	GET	"match%5B%5D=test_metric1%7Bfoo%3D~%22.%2Bo%22%7D&match%5B%5D=none"
        api_test.go:1157: run 16	POST	"match%5B%5D=test_metric1%7Bfoo%3D~%22.%2Bo%22%7D&match%5B%5D=none"
        api_test.go:1157: run 17	GET	"end=-1&match%5B%5D=test_metric2&start=-2"
        api_test.go:1157: run 17	POST	"end=-1&match%5B%5D=test_metric2&start=-2"
        api_test.go:1157: run 18	GET	"end=100001&match%5B%5D=test_metric2&start=100000"
        api_test.go:1157: run 18	POST	"end=100001&match%5B%5D=test_metric2&start=100000"
        api_test.go:1157: run 19	GET	"end=100000&match%5B%5D=test_metric2&start=-1"
        api_test.go:1157: run 19	POST	"end=100000&match%5B%5D=test_metric2&start=-1"
        api_test.go:1157: run 20	GET	"end=100&match%5B%5D=test_metric2&start=1"
        api_test.go:1157: run 20	POST	"end=100&match%5B%5D=test_metric2&start=1"
        api_test.go:1157: run 21	GET	"end=100000&match%5B%5D=test_metric2&start=1"
        api_test.go:1157: run 21	POST	"end=100000&match%5B%5D=test_metric2&start=1"
        api_test.go:1157: run 22	GET	"end=1&match%5B%5D=test_metric2&start=-1"
        api_test.go:1157: run 22	POST	"end=1&match%5B%5D=test_metric2&start=-1"
        api_test.go:1157: run 23	GET	""
        api_test.go:1157: run 23	POST	""
        api_test.go:1157: run 24	GET	""
        api_test.go:1157: run 25	GET	""
        api_test.go:1157: run 26	GET	"state=any"
        api_test.go:1157: run 27	GET	"state=active"
        api_test.go:1157: run 28	GET	"state=Dropped"
        api_test.go:1157: run 29	GET	"metric=go_threads"
        api_test.go:1157: run 30	GET	"match_target=%7Bjob%3D%22blackbox%22%7D"
        api_test.go:1157: run 31	GET	""
        api_test.go:1199: Response does not match, expected:
            [{"target":{"job":"test"},"metric":"go_threads","type":"gauge","help":"Number of OS threads created.","unit":""},{"target":{"job":"blackbox"},"metric":"prometheus_tsdb_storage_blocks_bytes","type":"gauge","help":"The number of bytes that are currently used for local storage by all blocks.","unit":""}]
            got:
            [{"target":{"job":"blackbox"},"metric":"prometheus_tsdb_storage_blocks_bytes","type":"gauge","help":"The number of bytes that are currently used for local storage by all blocks.","unit":""},{"target":{"job":"test"},"metric":"go_threads","type":"gauge","help":"Number of OS threads created.","unit":""}]
FAIL
FAIL	github.com/prometheus/prometheus/web/api/v1	0.062s
FAIL
```
**Environment**

* System information:
Linux 5.3.13-300.fc31.x86_64 x86_64

### PR DESCRIPTION
**Pull request by gotjosh**
Fixes flaky test for api/v1/targets/metadata.

Allows sorting of responses from the API. For our tests to be deterministic, we need to ensure the response from the API follows an order. This structure allows us to define one.

Fixes #6431

Signed-off-by: gotjosh <josue@grafana.com>

### CODE REVIEW

File: web/api/v1/api_test.go [hunk 1]
```diff
@@ -539,6 +540,7 @@ func testEndpoints(t *testing.T, api *API, testLabelAPI bool) {
 		query    url.Values
 		response interface{}
 		errType  errorType
+		sorter   func(interface{}) interface{}
```
Comment (brian-brazil) [inline web/api/v1/api_test.go]: A list of interfaces would be more in line with the sort package, and have it not return the result given that it mutates.

Comment (gotjosh) [reply web/api/v1/api_test.go]: I did consider that - the only problem with that it wouldn't match the response type we have right now resulting in a much bigger diff. Happy to amend it if you think it's still worth it!

Comment (gotjosh) [reply web/api/v1/api_test.go]: I have fixed the latter though.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**